### PR TITLE
1inch-U28: optimize main lineage — pre-materialize traces for looped queries

### PR DESCRIPTION
**Pull request by grkhr**

### CODE REVIEW

File: dex/models/_projects/oneinch/arbitrum/oneinch_arbitrum_ar_raw_traces.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{{
+    config(
+        schema = 'oneinch_' + blockchain,
+        alias = 'ar_raw_traces',
+        partition_by = ['block_date'],
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+        unique_key = ['tx_hash', 'trace_address']
```
Comment (jeff-dude) [inline dex/models/_projects/oneinch/arbitrum/oneinch_arbitrum_ar_raw_traces.sql:14]: another thing we can consider for performance gains:
can we add `block_date` to unique keys any time it's a partition column? that'll help trino reduce partitions to search for hash values on merge join conditions

File: macros/models/_project/oneinch/AR/oneinch_ar_raw_traces_macro.sql [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        {% if is_incremental() %}
+            {{ incremental_predicate('block_time') }}
+        {% else %}
+            block_time >= timestamp '{{date_from}}'
+        {% endif %}
+)
+
+
+select * from traces 
+join decoded_calls using(tx_hash, block_number, trace_address)
```
Comment (jeff-dude) [inline macros/models/_project/oneinch/AR/oneinch_ar_raw_traces_macro.sql:53]: is `block_date` an option to join on too?

Comment (jeff-dude) [reply macros/models/_project/oneinch/AR/oneinch_ar_raw_traces_macro.sql:53]: i think `block_number` would help, but `block_date` may be best option across most of dune

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: dex/models/_projects/oneinch/arbitrum/oneinch_arbitrum_ar_raw_traces.sql (hunk 1 above)
Comment (grkhr) [reply dex/models/_projects/oneinch/arbitrum/oneinch_arbitrum_ar_raw_traces.sql]: done

File: macros/models/_project/oneinch/AR/oneinch_ar_raw_traces_macro.sql (hunk 2 above)
Comment (grkhr) [reply macros/models/_project/oneinch/AR/oneinch_ar_raw_traces_macro.sql]: done

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**